models/_generate_layers.py

```diff
@@ -496,6 +497,38 @@ def collect_layers():
   return ls
 
 
+def get_super_call_params(layer):
+  """
+  Inspects the given layer, extracts unwanted parameters and then returns a string that is ready to be used in the super
+  call.
+  :param layer: Layer to be inspected
+  :return: string with super call parameters
+  """
+  # get code as string list
+  code = inspect.getsource(layer.__init__).splitlines()
+
+  # get list of lines starting with super and take the first one
+  lines = [line.strip() for line in code if line.strip().startswith("super")]
+  if not lines:
+    return ''
+
+  # reformat the super call to extract what we need
+  super_call = lines[0]
+  call = super_call.split(".")[1]
+  call_pruned = call[9:-1]
+
+  # remove excluded params
+  split = call_pruned.split(",")
+  tup_ls = list(map(lambda x: x.split("="), split))
+  for idx, param in enumerate(tup_ls):
+    if param[0].strip() in LayerSignature.IgnoreParamNames or param[0].strip() == "batch_norm":
```

Comment:
`batch_norm` should not be a hardcoded special case here.

Reply:
Then I am not sure how to handle this case. It is not in `IgnoreParamNames`, but removing it causes an error in `_generated_layers` for `BatchNorm`, because it is not defined there. Or do you suggest adding an additional list with "excluded parameters" which we then can modify/extend instead of just checking batchnorm? 
I guess I will add that, but if its about the "batch_norm" in general I am not sure how to handle.

Reply:
I don't really know which error you mean. Please be more specific.

Reply:
But in any case, this must not be a hardcoded extra case here. This should be more generic.

Reply:
In returnn_common we pass the parameters for Batch normalization explicitly via `get_opts` instead of giving them over (as in `RETURNN`) in the parameter `batch_norm`. This used to work, since we were not actually looking at the `super()` withing `RETURNN` which we are doing now.
What is not possible (without other changes I think) is adding `batch_norm` to excluded parameters, since that would interfere with the `_Base` layer.
Maybe I am overlooking something here, but right now I dont see a more generic way, but introducing a new list

Reply:
I think I would just add `batch_norm` to `_IgnoreParamNames`. I'm not sure it is a good idea to anyway use it directly on other layers. We do not need that here (and it might be even better to have it more clean).

Reply:
Adding it to _IgnoreParamNames excludes it from `_Base` layer too right? Or is there an exception for `_Base` somwhere I am missing?

Reply:
And I think in `_Base` it is necessary

Reply:
Oh yes you are right, my bad. Will be changed